Adding new getRow

tjpalmer85 opened the pull request:
Adding new function to return row as dict so we can access data directly for view history rather than as web element.

JS noob code so hopefully ok :)

Commit:
Adding new getRow

Adding new function to return row as dict so we can access data directly for view history rather than as web element

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,53 +61,44 @@ globalThis.PySeExt.GridHelper = {
+     * @param  {Object|Number} rowData      The index of or an object containing the row data for the record to be found.
+     * @return {store} The element for the found row.
+     */
+        getRowStore: function(gridSelector, rowData) {
```
This method is tabbed in too far.
Sorry, trivial, but consistency is good.

tjpalmer85 replied:
Tab indentation sorted

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,53 +61,44 @@ globalThis.PySeExt.GridHelper = {
+     * @param  {Object|Number} rowData      The index of or an object containing the row data for the record to be found.
+     * @return {store} The element for the found row.
+     */
+        getRowStore: function(gridSelector, rowData) {
```
Terminology thing, but the store is what is behind the entire grid.
This method gets the data for a row, so I think a better name is getRowData.

Make sure you update the comment too please.

tjpalmer85 replied:
all updated

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -181,5 +172,55 @@ globalThis.PySeExt.GridHelper = {
+     */
+    __findRowRecord: function(rowData, store) {
+        var 
+            rowRecord,
```
Move this up with the var please.

tjpalmer85 replied:
sorted

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -181,5 +172,55 @@ globalThis.PySeExt.GridHelper = {
+            foundIndex,
+            prop;
+
+                if (typeof(rowData) === 'number') {
```
This whole block is tabbed in too far.

tjpalmer85 replied:
Tab indentation sorted

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -181,5 +172,55 @@ globalThis.PySeExt.GridHelper = {
         return columnHeader;
+    },
+    /**
+     * Gets the row store with the specified data or index in the grid.
```
This returns the record not the store.

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,53 +61,44 @@ globalThis.PySeExt.GridHelper = {
-
-        return row;
-    },
+            return store.getAt(row).getData();
```
I think you need to guard against row being undefined or null here, since it might not have been found.

tjpalmer85 replied:
noob code added

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,53 +61,44 @@ globalThis.PySeExt.GridHelper = {
+                    store = grid.getStore();
 
-                if (foundIndex !== -1) {
-                    row = grid.getView().getRow(foundIndex);
```
The variable 'row' has not been declared meaning this creates a global!

tjpalmer85 replied:
have now declared

westy commented on pyseext/grid_helper.py:
```diff
@@ -399,6 +403,32 @@ def get_row(self, grid_cq: str, row_data: Union[int, dict], should_throw_excepti
 
         raise GridHelper.RowNotFoundException(grid_cq, row_data)
 
+    def get_row_store(self, grid_cq: str, row_data: Union[int, dict], should_throw_exception: bool = True) -> dict:
```
As discussed in the JS, this gets the data for a row, not a store.
Make sure update the template and comments as needed too please.

tjpalmer85 replied:
all updated

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,53 +61,44 @@ globalThis.PySeExt.GridHelper = {
-            foundIndex,
-            prop;
+        var me = this,
+        grids = globalThis.Ext.ComponentQuery.query(gridSelector),
```
This needs tabbing in please.

tjpalmer85 replied:
Tab indentation sorted

Commit:
Addressed code review comments

Changes based on code review comments.

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,52 +61,51 @@ globalThis.PySeExt.GridHelper = {
-                foundIndex = rowData;
+            row = me.__findRowRecord(rowData, store);
+        }
+        if (row !== undefined && row !== null) {
```
Can just use `if (row) {` since both undefined and null are falsey in JavaScript.

tjpalmer85 replied:
Updated accordingly

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,52 +61,51 @@ globalThis.PySeExt.GridHelper = {
+            return null;
+        }
+    },
+    /**
```
Need a space between methods please

tjpalmer85 replied:
Added

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -61,52 +61,51 @@ globalThis.PySeExt.GridHelper = {
-                }
+                row = me.__findRowRecord(rowData, store);
+            }
+            if (row !== undefined && row !== null) {
```
Again, use `if (row) {` please

tjpalmer85 replied:
Updated accordingly

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -181,5 +180,54 @@ globalThis.PySeExt.GridHelper = {
 
         return columnHeader;
+    },
+    /**
```
Need a space between the methods please.

Also, this method does not return a row record, but the index for the row.
It's name should be changed and header updated please.

tjpalmer85 replied:
Updated accordingly

Commit:
Few more changes following code review

Updated __findRowRecord to __findRowIndex and addressed some other code review comments

westy commented on pyseext/js/PySeExt.GridHelper.js:
```diff
@@ -181,5 +181,55 @@ globalThis.PySeExt.GridHelper = {
+     * The grid must be visible.
+     *
+     * @param  {Object|Number} rowData The index of or an object containing the row data for the record to be found.
+     * @return {Object} The data for the row record.
```
This needs updating, then I think that's it!
Something like:
`@return {Number} The index for the row record, or undefined if not found`

tjpalmer85 replied:
Ah bugger, sorry should have spotted that. Updated

Commit:
Updated function header as per comments
